modules/features/taskerplugin/src/main/java/au/com/shiftyjelly/pocketcasts/taskerplugin/controlplayback/config/ViewModelConfigControlPlayback.kt

```diff
@@ -16,76 +21,53 @@ class ViewModelConfigControlPlayback @Inject constructor(
-            shouldAskForState.tryEmit(askFor)
-        }
-
-        val valueState by lazy { MutableStateFlow(valueGetter(input)) }
-        var value: String? = null
-            set(value) {
-                valueSetter(value)
-                valueState.tryEmit(value)
-            }
+        override val askFor get() = showForCommands.isEmpty() || showForCommands.contains(input?.commandEnum)
     }
 
-    /**
-     * When adding new optional fields here make sure to add them to the [optionalFields] list below so that everything is correctly updated
-     */
-    private val optionalFieldChapterToSkipTo by lazy { OptionalField(InputControlPlayback.PlaybackCommand.SkipToChapter, { input?.chapterToSkipTo }, { input?.chapterToSkipTo = it }) }
-    private val optionalFieldTimeToSkipTo by lazy { OptionalField(InputControlPlayback.PlaybackCommand.SkipToTime, { input?.skipToSeconds }, { input?.skipToSeconds = it }) }
-    private val optionalFieldTimeToSkip by lazy { OptionalField(listOf(InputControlPlayback.PlaybackCommand.SkipForward, InputControlPlayback.PlaybackCommand.SkipBack), { input?.skipSeconds }, { input?.skipSeconds = it }) }
-    private val optionalFields = listOf(
-        optionalFieldChapterToSkipTo,
-        optionalFieldTimeToSkipTo,
-        optionalFieldTimeToSkip
-    )
-
-    val commandState by lazy {
-        MutableStateFlow(
-            input?.command
-        )
+    private val inputFieldCommand = object : InputField<InputControlPlayback.PlaybackCommand>(R.string.playback_command, drawable.filter_play, { command }, { setCommand(it) }) {
+        override fun getPossibleValues() = MutableStateFlow(InputControlPlayback.PlaybackCommand.values().toList())
+        override fun getValueDescriptionSpecific(possibleValue: InputControlPlayback.PlaybackCommand?) = possibleValue?.getDescription(context)
     }
-    var command: String?
-        get() = input?.command
-        set(value) {
-            input?.command = value
-            commandState.tryEmit(value)
-            optionalFields.forEach { it.updateAskForState() }
-        }
+    override val inputFields: List<InputFieldBase<*>> = listOf(
+        inputFieldCommand,
+        InputField(R.string.chapter_to_skip_to, drawable.filter_bullet, InputControlPlayback.PlaybackCommand.SkipToChapter, { chapterToSkipTo }, { chapterToSkipTo = it }),
+        InputField(R.string.time_to_skip_to_seconds, drawable.filter_time, InputControlPlayback.PlaybackCommand.SkipToTime, { skipToSeconds }, { skipToSeconds = it }),
+        InputField(R.string.time_to_skip_seconds, drawable.filter_time, listOf(InputControlPlayback.PlaybackCommand.SkipForward, InputControlPlayback.PlaybackCommand.SkipBack), { skipSeconds }, { skipSeconds = it }),
+        object : InputField<Double>(R.string.playback_speed_between_values, au.com.shiftyjelly.pocketcasts.taskerplugin.R.drawable.speedometer, InputControlPlayback.PlaybackCommand.SetPlaybackSpeed, { playbackSpeed }, { playbackSpeed = it }) {
```

Comment:
Can we avoid the fully qualified name here: `au.com.shiftyjelly.pocketcasts.taskerplugin.R.drawable.speedometer`? My suggestion would be to update your imports along these lines:

```
import au.com.shiftyjelly.pocketcasts.taskerPlugin.R
import au.com.shiftyjelly.pocketcasts.images.R as IR
import au.com.shiftyjelly.pocketcasts.localization.R as LR
```

Reply:
Thank you! Will fix when I push another commit.